core/drivers/crypto/hisilicon/hpre_ecc.c

```diff
@@ -974,12 +1023,467 @@ static TEE_Result hpre_ecc_do_shared_secret(struct drvcrypt_secret_data *sdata)
 	return ret;
 }
 
+static enum hisi_drv_status
+hpre_ecc_sign_params_fill(const struct hpre_ecc_curve *curve,
+			  struct hpre_ecc_msg *msg,
+			  struct drvcrypt_sign_data *sdata,
+			  struct bignum *rand_k)
+{
+	struct ecc_keypair *ecc_key = (struct ecc_keypair *)sdata->key;
+	struct hpre_ecc_sign *ecc_sign = &msg->param_size.ecc_sign;
+	uint8_t *p = msg->key;
+	uint8_t *a = p + msg->key_bytes;
+	uint8_t *d = a + msg->key_bytes;
+	uint8_t *b = d + msg->key_bytes;
+	uint8_t *n = b + msg->key_bytes;
+	uint8_t *gx = n + msg->key_bytes;
+	uint8_t *gy = gx + msg->key_bytes;
+	uint8_t *e = msg->in;
+	uint8_t *k = e + msg->key_bytes;
+	enum hisi_drv_status ret = HISI_QM_DRVCRYPT_NO_ERR;
+
+	memcpy(p, curve->p, msg->curve_bytes);
+	memcpy(a, curve->a, msg->curve_bytes);
+	crypto_bignum_bn2bin(ecc_key->d, d);
+	ecc_sign->d_bytes = crypto_bignum_num_bytes(ecc_key->d);
+	memcpy(b, curve->b, msg->curve_bytes);
+	memcpy(n, curve->n, msg->curve_bytes);
+	memcpy(gx, curve->x, msg->curve_bytes);
+	memcpy(gy, curve->y, msg->curve_bytes);
+	crypto_bignum_bn2bin(rand_k, k);
+	ecc_sign->k_bytes = crypto_bignum_num_bytes(rand_k);
+
+	ecc_sign->e_bytes = (sdata->message.length > msg->curve_bytes ?
```

Comment:
Please use an if statement instead of the ternary operator.
Edit: or even better:
``` C
ecc_sign->e_bytes = MIN(sdata->message.length, msg->curve_bytes);
```